libs/blocks/global-navigation/utilities/getUserEntitlements.js

```diff
@@ -36,7 +37,75 @@ const getQueryParameters = (params) => {
   return result;
 };
 
+const emptyEntitlements = {
+  clouds: {},
+  arrangment_codes: {},
+  fulfilled_codes: {},
+  offer_families: {},
+  list: { fulfilled_codes: [] },
+};
+const CREATIVE_CLOUD = 'creative_cloud';
+const DOCUMENT_CLOUD = 'document_cloud';
+const EXPERIENCE_CLOUD = 'experience_cloud';
+
+/**
+ * Breaks JIL offers(subscriptions) into easily addressable flat data structure.
+ * @param {*} offers
+ */
+const mapSubscriptionCodes = (offers) => {
+  if (!Array.isArray(offers)) {
+    return emptyEntitlements;
+  }
+
+  const {
+    clouds,
+    arrangment_codes,
+    fulfilled_codes,
+    offer_families,
+    list,
+  } = emptyEntitlements;
+
+  offers.forEach(({ fulfilled_items, offer = {} }) => {
+    const cloud = offer.product_arrangement?.cloud;
+    clouds[CREATIVE_CLOUD] = clouds[CREATIVE_CLOUD] || cloud === 'CREATIVE';
+    clouds[DOCUMENT_CLOUD] = clouds[DOCUMENT_CLOUD] || cloud === 'DOCUMENT';
+    clouds[EXPERIENCE_CLOUD] = clouds[EXPERIENCE_CLOUD] || cloud === 'EXPERIENCE';
+
+    const family = offer.product_arrangement?.family;
+    if (family) {
+      offer_families[family.toLowerCase()] = true;
+    }
+
+    if (offer.product_arrangement_code) {
+      arrangment_codes[offer.product_arrangement_code] = true;
+    }
+
+    if (Array.isArray(fulfilled_items)) {
+      fulfilled_items.forEach(({ code }) => {
+        if (code) {
+          fulfilled_codes[code] = true;
+          // Avoid duplicates
+          if (list.fulfilled_codes.indexOf(code) === -1) {
+            list.fulfilled_codes.push(code);
+          }
+        }
+      });
+    }
+  });
+
+  return {
+    clouds,
+    arrangment_codes,
+    fulfilled_codes,
+    offer_families,
+    list,
+  };
+};
+
 const getSubscriptions = async ({ queryParams, locale }) => {
+  const controller = new AbortController();
+  const { signal } = controller;
+  const timeout = setTimeout(() => controller.abort(), API_WAIT_TIMEOUT);
```

Comment:
Maybe it makes more sense to use `AbortSignal.timeout(API_WAIT_TIMEOUT)` instead of setting a custom timeout? See https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/timeout